### PR TITLE
Add FileList component to Intervention details

### PR DESCRIPTION
**Pull request by CorneeSean**
~~First commit: Handle empty attachment list
Second commit: Programatically trigger file download - this was needed since for simple anchor link click, the request was rejected with 401 unauthorized by Cloud storage. x-functions-key needs to be passed in request header, hence the need to move to XHR request and code triggering the download~~

Adds styled file list component to intervention details

### CODE REVIEW

File: frontend/src/app/modules/interventions/details/details.component.html [hunk 1]
```diff
@@ -40,7 +40,7 @@ <h2 class="mat-h2">Załączniki</h2>
       <!-- TODO:  Temporary solution - UX-friendly attachment list with download option needed in the near future -->
       <p *ngIf="!intervention.attachments || !intervention.attachments.length">Brak załączników</p>
       <p *ngFor="let file of intervention.attachments">
-        <a href="{{ createFileUrl(file.id) }}" rel="noopener noreferer" download>
+        <a href="#" (click)="fileDownload.emit(file); $event.preventDefault()">
```
Comment (chodorowicz) [inline frontend/src/app/modules/interventions/details/details.component.html:43]: I wouldn't create it as a `<a>` tag which looks like a like. People are use to clicking cmd or right clicking and selecting "open in a new tab" 
Maybe plain name and download `<button>` next to it. 
image

Comment (CorneeSean) [reply frontend/src/app/modules/interventions/details/details.component.html:43]: I agree Kuba. This is temporary and will be replaced in my next PR. Main point of this PR is the logic and mechanics in service to download a file - not a UI representation

File: frontend/src/app/modules/interventions/details/details.component.ts [hunk 2]
```diff
@@ -1,31 +1,29 @@
...
 @Component({
-  selector: 'app-intervention-details',
+  selector: 'eko-intervention-details',
   templateUrl: './details.component.html',
   styleUrls: ['./details.component.scss'],
 })
 export class DetailsComponent extends ComponentWithSubscriptions {
   @Input() intervention: Intervention;
   @Input() embedded?: boolean;
+  @Output() fileDownload = new EventEmitter<Attachment>();
```
Comment (CorneeSean) [inline frontend/src/app/modules/interventions/details/details.component.ts:18]: Delete

**Comment by CorneeSean** [comment]
@chodorowicz please review once again in your free time, as I've pushed changes on top of the previous ones. This is adding the actual file list component with buttons you mentioned :) Nope, I am not writing that fast - I almost had it ready anyway ;)

File: frontend/src/app/shared/utils/file.utils.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    if (size < 1024 * 500) {
+      // up to 500KB
+      return ((size / 1024) | 0) + 'KB'; // tslint:disable-line
+    }
+    if (size < 1024 * 1024 * 100) {
+      // up to 100MB
+      return (size / (1024 * 1024)).toPrecision(2) + 'MB';
+    }
+    if (size < 1024 * 1024 * 1024) {
+      // up to 10MB
```
Comment (CorneeSean) [inline frontend/src/app/shared/utils/file.utils.ts:25]: The comment doesn't seem to be accurate

Comment (chodorowicz) [reply frontend/src/app/shared/utils/file.utils.ts:25]: 👍

**Pull request merged**